File: LeetCode/Easy/121 - Best Time to Buy and Sell Stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let res = 0, pre = prices[0];
    for (let i of prices) {
        pre = Math.min(pre, i);
        res = Math.max(res, i - pre);
    }
    return res;
};

File: LeetCode/Easy/121 - Best Time to Buy and Sell Stock/121 - Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mn = prices[0], ret = 0;
        for (auto price : prices) {
            mn = min(mn, price);
            
            ret = max(ret, price - mn);
        }
        return ret;
    }
};